x/dualstaking/keeper/delegator_reward.go

```diff
@@ -92,3 +94,43 @@ func (k Keeper) CalcDelegationsSum(stakeEntry epochstoragetypes.StakeEntry) math
 func (k Keeper) CalcDelegatorReward(delegatorsReward math.Int, totalDelegations math.Int, delegation types.Delegation) math.Int {
 	return delegatorsReward.Mul(delegation.Amount.Amount).Quo(totalDelegations)
 }
+
+func (k Keeper) ClaimRewards(ctx sdk.Context, delegator string, provider string) error {
+	goCtx := sdk.WrapSDKContext(ctx)
+	res, err := k.DelegatorRewards(goCtx, &types.QueryDelegatorRewardsRequest{Delegator: delegator, Provider: provider})
+	if err != nil {
+		return utils.LavaFormatWarning("could not claim delegator rewards", err,
+			utils.Attribute{Key: "delegator", Value: delegator},
+		)
+	}
+
+	for _, reward := range res.Rewards {
+		delegatorAcc, err := sdk.AccAddressFromBech32(delegator)
+		if err != nil {
+			utils.LavaFormatError("could not claim delegator reward from provider", err,
```

Comment:
this is really bad, should be "critical"

Reply:
done. see https://github.com/lavanet/lava/commit/302b2bec531bd9308d142d746c327509bde9b547